### PR TITLE
Add ObjectSerializer and JsonObjectSerializer to Azure.Experimental

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs [hunk 1]
```diff
@@ -1,3 +1,23 @@
+namespace Azure.Core
+{
+    public partial class JsonObjectSerializer : Azure.Core.ObjectSerializer
+    {
+        public JsonObjectSerializer() { }
+        public JsonObjectSerializer(System.Text.Json.JsonSerializerOptions options) { }
```
Comment (tg-msft) [inline sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs:6]: I wonder if there's any reason we should consider abstracting away our own set of JsonSerializerOptions.   We can save that question until we figure out if/how we support other JSON serializers.

File: sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        /// <param name="options">The <see cref="JsonSerializerOptions"/> instance to use when serializing/deserializing.</param>
+        public JsonObjectSerializer(JsonSerializerOptions options)
+        {
+            _options = options;
+        }
+
+        /// <inheritdoc />
+        public override void Serialize(Stream stream, object? value, Type inputType)
+        {
+            var buffer = Encoding.UTF8.GetBytes(JsonSerializer.Serialize(value, inputType, _options));
```
Comment (tg-msft) [inline sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs:38]: Should we use https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serializetoutf8bytes?view=netcore-3.1 instead?

**Review by tg-msft — APPROVED** [review]
Looks great

File: sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs (hunk 1 above)
Comment (pakrym) [reply sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.netstandard2.0.cs:6]: There is a tentative plan to ship a Newtonsoft based implementation of this.

File: sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs (hunk 2 above)
Comment (pakrym) [reply sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs:38]: Yup.

**Pull request merged**